src/AutomatedCar/SystemComponents/Sensors/Sensor.cs

```diff
@@ -1,30 +1,44 @@
 ﻿namespace AutomatedCar.SystemComponents.Sensors
 {
     using System.Collections.Generic;
+    using System.Linq;
+    using System.Numerics;
     using AutomatedCar.Models;
     using AutomatedCar.SystemComponents.Packets;
 
-    internal abstract class Sensor : SystemComponent
+    public abstract class Sensor : SystemComponent
     {
         protected ISensorPacket sensorPacket;
 
+        protected Vector2 X { get; set; }
+
+        protected Vector2 Y { get; set; }
+
+        protected Vector2 Z { get; set; }
+
         public Sensor(VirtualFunctionBus virtualFunctionBus)
```

Comment:
Kiszámolhatnánk Konstruktorban Angle of Viewet (https://en.wikipedia.org/wiki/Angle_of_view) Viewing Distance és Viewing Angle alapján, azzal a matekkal, amit papíron is használtál. Ezáltal 1. Egyértelmű, hogyan inicializálod a háromszöget, 2. Eltűnnek a Magic Numberek. OFC lassaabb lesz a program tőle, ha akadozik ezt vissza lehet változtatni.